src/streamlit_extras/__init__.py

```diff
@@ -0,0 +1,45 @@
+import inspect
+from importlib import import_module
+from typing import Any, Callable, Optional, TypeVar, Union, overload
+
+from streamlit import _gather_metrics
+
+F = TypeVar("F", bound=Callable[..., Any])
+
+
+@overload
+def extra(
+    func: F,
+) -> F:
+    ...
+
+
+@overload
+def extra(
+    func: None = None,
+) -> Callable[[F], F]:
+    ...
+
+
+def extra(
+    func: Optional[F] = None,
+) -> Union[Callable[[F], F], F]:
+
+    if func:
+
+        filename = inspect.stack()[1].filename
+        extra_name = "streamlit_extras." + filename.split("/")[-2]
+        module = import_module(extra_name)
```

Comment:
nit: Wouldn't it be possible to directly determine the module from `func`, e.g.: `inspect.getmodule(func)`?

Reply:
Hehe so that's what I did first. But since some of these extras are actually referring to external packages, the guessed module was one outside streamlit_extras project. Even while passing the _filename argument apparently... I agree this is hacky-- but works. What do you think?

Reply:
Ahh I see, hmm maybe you can directly get the module instance from: `inspect.currentframe()` but otherwise I don't have any good ideas on how to do it better...

Reply:
hmm, but there could be an issue with different uses: `@extra()` vs `@extra` ... because the call stack is different

Reply:
Perhaps a slightly-less-hacky version would just be `"streamlit_extras." + func.__module__`

Reply:
@blackary func.__module__ would give me streamlit_extras back. Keeping it as is for now